docs/389ds/design/audit-json-logging-design.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Audit JSON Logging Design"
+---
+
+# Audit JSON Logging
+----------------
+
+Overview
+--------
+
+In order to make the audit logging more consumable by standard parsing tool it will written in a JSON format instead of using a partial LDIF format which not LDAP compliant in its current form.
+
+While the old format, after a bit if data massaging, could be replayed using a LDAP client (e.g. ldapmodify), this is most likely not used by most customers.  Instead, customers are looking for actual audit information that can by consumed by well known log parsers.
+
+JSON Design
+------------
+
+```
+[
+    {
+        date: 20243457458475847Z
+        target_dn: DN,
+        bind_dn: DN,
+        client: IP_ADDRESS,
```

Comment:
Now that we support HAProxy protocol, we can log both the proxy and the client's real IP addresses, see for example https://github.com/389ds/389-ds-base/blob/b29560435fefd67360ba2eaff0d9f6bb8ddde0b3/ldap/servers/slapd/connection.c#L1300-L1302
How it will be reflected in the audit log?

Reply:
Ah yes good point about HAProxy. Normally I would only log the IP address that we log in the access log, but yeah we can log both it the audit log, and when we redo the access log we can display both there as well.

Reply:
Sounds good.